### ISSUE TITLE
doc_include_without_cfg "Added in" version is wrong

### ISSUE THREAD

**Issue by Xiretza** [rust-lang/rust-clippy#14135]
### Summary

https://rust-lang.github.io/rust-clippy/master/index.html#doc_include_without_cfg says "Added in: 1.84.0", but clippy from the rustup 1.84.1 toolchain (`cargo clippy --version` says `clippy 0.1.84 (9fc6b43126 2025-01-07)`) doesn't know the lint.

### Reproducer

I tried this code:

```rust
#![warn(clippy::doc_include_without_cfg)]
```

I expected to see this happen:

No output

Instead, this happened:

```
warning: unknown lint: `clippy::doc_include_without_cfg`
 --> src/main.rs:1:9
  |
1 | #![warn(clippy::doc_include_without_cfg)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unknown_lints)]` on by default
```

### Version

```text
rustc 1.84.1 (e71f9a9a9 2025-01-27)
binary: rustc
commit-hash: e71f9a9a98b0faf423844bf0ba7438f29dc27d58
commit-date: 2025-01-27
host: x86_64-unknown-linux-gnu
release: 1.84.1
LLVM version: 19.1.5
```

**Comment by alex-semenyuk** [comment]
Can be closed, now version 1.85.0
